src/dmd/iasmdmd.d

```diff
@@ -1253,25 +1253,24 @@ code *asm_emit(Loc loc,
     uint  uSizemaskTmp;
     const(REG) *pregSegment;
     //ASM_OPERAND_TYPE    aopty1 = _reg , aopty2 = 0, aopty3 = 0;
-    ASM_MODIFIERS       amod1 = _normal, amod2 = _normal;
-    uint            uSizemaskTable1 =0, uSizemaskTable2 =0,
-                        uSizemaskTable3 =0;
-    ASM_OPERAND_TYPE    aoptyTable1 = _reg, aoptyTable2 = _reg, aoptyTable3 = _reg;
-    ASM_MODIFIERS       amodTable1 = _normal,
-                        amodTable2 = _normal;
-    uint            uRegmaskTable1 = 0, uRegmaskTable2 =0;
+    ASM_MODIFIERS[2] amods = _normal;
+    uint[3] uSizemaskTable = 0;
+    ASM_OPERAND_TYPE[3] aoptyTable = _reg;
+    ASM_MODIFIERS[2] amodTable = _normal;
```

Comment:
What is the point of this "array-ification"?

Reply:
They allow factoring setRegisterProperties.  I hoped I'd find more places where repeated code could turn into loops or functions (I suspect there are more but this PR was getting a bit big anyway).  I also think logically they should be arrays as they are a series of related values